### PR TITLE
refs #17114: workaround for arm64 which should segfault but does not

### LINKED ISSUE #17114: std/segfaults doesn't raise (reliably?) NilAccessDefect on ARM64 Linux nor windows
### Build log
See full log [here](https://github.com/nim-lang/nightlies/runs/1939486935?check_suite_focus=true#step:9:5906)

```
   Hint: /home/runner/work/nightlies/nightlies/nim-1.5.1/build/nimcache/nim/wrapnils_r/runnableExamples/wrapnils_group0_examples  [Exec]
  0
  /home/runner/work/nightlies/nightlies/nim-1.5.1/lib/system/assertions.nim(125) wrapnils_examples1
  /home/runner/work/nightlies/nightlies/nim-1.5.1/lib/system/assertions.nim(58) raiseAssert
  /home/runner/work/nightlies/nightlies/nim-1.5.1/lib/system/fatal.nim(53) sysFatal
  Error: unhandled exception: expected raising 'NilAccessDefect', instead nothing was raised by: 
  echo (?.f2.x2.x2).x3[] [AssertionDefect]
  Error: execution of an external program failed: '/home/runner/work/nightlies/nightlies/nim-1.5.1/build/nimcache/nim/wrapnils_r/runnableExamples/wrapnils_group0_examples '
  [runnableExamples] failed: generated file: '/home/runner/work/nightlies/nightlies/nim-1.5.1/build/nimcache/nim/wrapnils_r/runnableExamples/wrapnils_group0_examples.nim' group: 'rdoccmd: 
  docCmd: 
  code: # autogenerated by docgen
  # source: /home/runner/work/nightlies/nightlies/nim-1.5.1/lib/std/wrapnils.nim
  # rdoccmd: 
  import r"/home/runner/work/nightlies/nightlies/nim-1.5.1/build/nimcache/nim/wrapnils_r/runnableExamples/wrapnils_examples1.nim"
  
  index: 0
  ' cmd: /home/runner/work/nightlies/nightlies/nim-1.5.1/bin/nim c -r --lib:/home/runner/work/nightlies/nightlies/nim-1.5.1/lib --warning:UnusedImport:off --path:/home/runner/work/nightlies/nightlies/nim-1.5.1/lib/std --nimcache:/home/runner/work/nightlies/nightlies/nim-1.5.1/build/nimcache/nim/wrapnils_r/runnableExamples   /home/runner/work/nightlies/nightlies/nim-1.5.1/build/nimcache/nim/wrapnils_r/runnableExamples/wrapnils_group0_examples.nim
  
  Took 71m16.120s
  Error: Process completed with exit code 1.
```

### Additional Information

First happened with [Nightlies build #205](https://github.com/nim-lang/nightlies/runs/1859390226?check_suite_focus=true#step:9:4242), commit ba64d0c8ab93c7f38b87567059cb2b66747d022c.

Commit: 1efaef52a2f916441d153917486ef8a4218a6111

## links
related:
* [SIGSEGV hangs instead of raising NilAccessDefect with `import segfaults` and gc:arc · Issue #16415 · nim-lang/Nim](https://github.com/nim-lang/Nim/issues/16415)
* [memory corrupted; possibly by try/catch using `import segfaults` · Issue #9752 · nim-lang/Nim](https://github.com/nim-lang/Nim/issues/9752)
* (somewhat related) [setControlCHook - Nim forum](https://forum.nim-lang.org/t/7886)

### PR DESCRIPTION
**Pull request by timotheecour**
refs #17114

### CODE REVIEW

**Review by ringabout — APPROVED** [review]

**Comment by Araq** [comment]
I don't agree that `wrapnils` should even mention `std / segfaults`, even if I were to agree about its usefulness, examples shouldn't show unrelated constructs.

File: lib/std/wrapnils.nim [hunk 1]
```diff
@@ -24,7 +24,10 @@ runnableExamples:
 
   assert (?.f2.x2.x2).x3 == nil  # this terminates ?. early
   import std/segfaults # enable `NilAccessDefect` exceptions
-  doAssertRaises(NilAccessDefect): echo (?.f2.x2.x2).x3[]
+  when not defined(arm64):
+    doAssertRaises(NilAccessDefect): echo (?.f2.x2.x2).x3[]
```
Comment (Araq) [inline lib/std/wrapnils.nim]:
```suggestion
```

File: lib/std/wrapnils.nim [hunk 2]
```diff
@@ -24,7 +24,8 @@ runnableExamples:
 
   assert (?.f2.x2.x2).x3 == nil  # this terminates ?. early
   import std/segfaults # enable `NilAccessDefect` exceptions
-  doAssertRaises(NilAccessDefect): echo (?.f2.x2.x2).x3[]
+
+# xxx pending bug #17178, un-disable `arm64`.
```
Comment (Araq) [inline lib/std/wrapnils.nim]:
```suggestion
```

File: lib/std/wrapnils.nim [hunk 3]
```diff
@@ -24,7 +24,6 @@ runnableExamples:
 
   assert (?.f2.x2.x2).x3 == nil  # this terminates ?. early
   import std/segfaults # enable `NilAccessDefect` exceptions
```
Comment (ringabout) [inline lib/std/wrapnils.nim:26]: then this line should be removed.

Comment (ringabout) [reply lib/std/wrapnils.nim:26]: fixed

**Pull request merged**